### ISSUE TITLE
proposal: "strictly fixed" versioning mode

### ISSUE THREAD

**Issue by dlqqq** [lerna/lerna#3569]
- [x] I'd be willing to implement this feature ([contributing guide](https://github.com/lerna/lerna/blob/main/CONTRIBUTING.md))

## Description

The current fixed versioning mode is confusing because it doesn't actually keep project versions fixed when bumping versions via `lerna version x.y.z`. From the documentation:

> if a module has been updated since the last time a release was made, it will be updated to the new version you're releasing. This means that you only publish a new version of a package when you need to

What this actually means is that it only bumps the version when it detects changes in that project. This is doubly confusing because it seems to contradict the sentence after that paragraph:

> Use this if you want to automatically tie all package versions together.

So, fixed versioning mode isn't actually fixed, and doesn't actually tie all package versions together! That seems preposterous. If you have two projects, `project/a` and `project/b` both at `1.0.0`, doing `lerna version major` after making changes exclusively to `project/b` causes `project/a` to remain versioned `1.0.0` while causing `project/b` to be versioned `2.0.0`.

Keep in mind that nothing functional is breaking; but it's much easier to reason about intra-monorepo compatibility when all versions are in lock-step with one another.

**I propose a more simply "strict fixed" versioning mode that simply bumps all of the versions across all projects regardless of whether files were changed.** 

## Motivation

- There already exists demonstrated developer demand for this. See this issue (by the author of Slate, @ianstormtaylor), closed due to staleness: https://github.com/lerna/lerna/issues/2369

- The current behavior of the fixed versioning mode is confusing because it is a misnomer, and the documentation could easily be misinterpreted to be self-contradictory.

- A strict fixed versioning mode makes it much easier for developers and users to reason about intra-monorepo package compatibility. I don't need to consult the Git history and keep some "compatibility document" updated every release. I just say `@project/a 1.0.0` is only compatible with `@project/b 1.0.0`. No promises beyond that.

## Suggested Implementation

I suggest a `--strict/--strict-fixed` flag passed to `lerna version` that would enable users to use the "strict fixed" versioning strategy outlined above. I'm not sure if this would also go on `lerna publish` as I have never used it, but it seems likely.

## Alternate Implementations

Open to discussion.

**Comment by fahslaj** [comment]
Hi @dlqqq , thank you for opening this feature request. It is a good call out that the documentation does not make this behavior clear.


I will note that you can use the `--force-publish` option to force Lerna to publish all packages, regardless if they have changed. By setting this option in `lerna.json`, you can achieve the behavior you are looking for. Lerna will always bump the version of new packages to whatever is in `lerna.json` and publish all of them at once, every time.

**Comment by dlqqq** [comment]
@fahslaj Yeah, you're right, I just reproduced it locally. I would have never thought of trying `--force-publish` without your guidance. Thank you so much! ❤️ 

This issue should be reinterpreted as a documentation / CLI syntax issue rather than a feature request.
